pkgs/tools/package-management/libdnf/default.nix

```diff
@@ -0,0 +1,62 @@
+{ gcc9Stdenv, stdenv, fetchFromGitHub, cmake, gettext, pkgconfig, gpgme, libsolv, openssl, check, pcre, json_c, libmodulemd, libsmartcols, sqlite, librepo, libyaml, rpm }:
+
+gcc9Stdenv.mkDerivation rec {
+  pname = "libdnf";
+  version = "0.55.2";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "0hiydwfa90nsrqk5ffa6ks1g73wnsgjgq7z7gwq9jj76a7gpfbfq";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    pkgconfig
+    check
+  ];
+
+  propagatedBuildInputs = [
+    gpgme
+    openssl
+    json_c
+    sqlite
+    librepo
+    pcre
+    rpm
+    libsolv
+    libsmartcols
+    libyaml
+    libmodulemd
+  ];
+
+  patches = stdenv.lib.optionals stdenv.isDarwin [ ./darwin.patch ];
+
+  prePatch = ''
+    # According to #cmake on freenode, libdnf should bundle the FindLibSolv.cmake module
+    cp ${libsolv}/share/cmake/Modules/FindLibSolv.cmake cmake/modules/
+  '';
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
```

Comment:
Why disable the testing? You might need `-DCMAKE_SKIP_BUILD_RPATH=OFF` but otherwise `doCheck = true;` should be preferred if possible.

Reply:
If I don't do that, I end up with this (after setting):

```
[100%] Linking CXX executable run_tests
Undefined symbols for architecture x86_64:
  "CppUnit::SourceLine::SourceLine(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)", referenced from:
      ModuleProfileTest::testGetName()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetDescription()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetContent()       in ModuleProfileTest.cpp.o
      ModulePackageTest::testSimpleGetters()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetArtifacts()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetProfiles()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetModuleDependencies()       in ModulePackageTest.cpp.o
      ...
  "CppUnit::TextTestRunner::run(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool, bool)", referenced from:
      _main in run_tests.cpp.o
  "CppUnit::AdditionalMessage::AdditionalMessage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in DependencyContainerTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      void CppUnit::assertEquals<libdnf::TransactionItemReason>(libdnf::TransactionItemReason const&, libdnf::TransactionItemReason const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      ...
  "CppUnit::CompilerOutputter::CompilerOutputter(CppUnit::TestResultCollector*, std::basic_ostream<char, std::char_traits<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      _main in run_tests.cpp.o
  "CppUnit::TestFactoryRegistry::getRegistry(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      __GLOBAL__sub_I_ModuleProfileTest.cpp in ModuleProfileTest.cpp.o
      __GLOBAL__sub_I_ModulePackageTest.cpp in ModulePackageTest.cpp.o
      __GLOBAL__sub_I_ContextTest.cpp in ContextTest.cpp.o
      __GLOBAL__sub_I_ModulePackageContainerTest.cpp in ModulePackageContainerTest.cpp.o
      __GLOBAL__sub_I_PackageTest.cpp in PackageTest.cpp.o
      __GLOBAL__sub_I_DependencyTest.cpp in DependencyTest.cpp.o
      __GLOBAL__sub_I_DependencyContainerTest.cpp in DependencyContainerTest.cpp.o
      ...
  "CppUnit::Message::addDetail(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      TransactionTest::testInsert()       in TransactionTest.cpp.o
      TransactionTest::testInsertWithSpecifiedId()       in TransactionTest.cpp.o
      TransactionTest::testInsert() [clone .cold] in TransactionTest.cpp.o
      TransactionTest::testInsertWithSpecifiedId() [clone .cold] in TransactionTest.cpp.o
  "CppUnit::Message::Message(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      TransactionTest::testInsert()       in TransactionTest.cpp.o
      TransactionTest::testInsertWithSpecifiedId()       in TransactionTest.cpp.o
  "CppUnit::Message::Message(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      ModuleProfileTest::testGetName()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetDescription()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetContent()       in ModuleProfileTest.cpp.o
      ModulePackageTest::testSimpleGetters()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetArtifacts()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetProfiles()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetModuleDependencies()       in ModulePackageTest.cpp.o
      ...
  "CppUnit::Message::Message(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      MergedTransactionTest::testMerge()      in MergedTransactionTest.cpp.o
  "CppUnit::Asserter::failNotEqual(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, CppUnit::SourceLine const&, CppUnit::AdditionalMessage const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)", referenced from:
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in DependencyContainerTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      void CppUnit::assertEquals<libdnf::TransactionItemReason>(libdnf::TransactionItemReason const&, libdnf::TransactionItemReason const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      ...
  "CppUnit::TestCase::TestCase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      ModuleProfileTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModuleProfileTest.cpp.o
      ModulePackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      ModulePackageContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageContainerTest.cpp.o
      PackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in PackageTest.cpp.o
      DependencyTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyTest.cpp.o
      DependencyContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyContainerTest.cpp.o
      ...
  "CppUnit::TestSuite::TestSuite(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)", referenced from:
      CppUnit::TestSuiteFactory<ModuleProfileTest>::makeTest()       in ModuleProfileTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageTest>::makeTest()       in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageContainerTest>::makeTest()       in ModulePackageContainerTest.cpp.o
      CppUnit::TestSuiteFactory<PackageTest>::makeTest()       in PackageTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyTest>::makeTest()       in DependencyTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyContainerTest>::makeTest()       in DependencyContainerTest.cpp.o
      ...
  "CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const", referenced from:
      ModuleProfileTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModuleProfileTest.cpp.o
      ModulePackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      ModulePackageContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageContainerTest.cpp.o
      PackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in PackageTest.cpp.o
      DependencyTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyTest.cpp.o
      DependencyContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyContainerTest.cpp.o
      ...
  "CppUnit::Test::findTestPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::TestPath&) const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::Test::resolveTestPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::Test::findTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::TestCase::getName[abi:cxx11]() const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::TestNamer::getFixtureName[abi:cxx11]() const", referenced from:
      CppUnit::TestSuiteFactory<ModuleProfileTest>::makeTest()       in ModuleProfileTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageTest>::makeTest()       in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageContainerTest>::makeTest()       in ModulePackageContainerTest.cpp.o
      CppUnit::TestSuiteFactory<PackageTest>::makeTest()       in PackageTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyTest>::makeTest()       in DependencyTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyContainerTest>::makeTest()       in DependencyContainerTest.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/run_tests.dir/build.make:368: tests/run_tests] Error 1
make[1]: *** [CMakeFiles/Makefile2:833: tests/CMakeFiles/run_tests.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

Reply:
Could you open an issue upstream and add a comment with a link here, the?

Reply:
I am not quite sure if this is a cppunit problem, a libdnf problem, a nix problem or a cmake problem.

If I had to guess it's cppunit

Reply:
I guess a nixpkgs issue tracker is the best place for now.

Reply:
https://github.com/NixOS/nixpkgs/issues/107428